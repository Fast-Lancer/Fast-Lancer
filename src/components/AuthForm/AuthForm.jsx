import { useState } from 'react'
import Button from '../Button/Button.jsx'
import styles from './AuthForm.module.css'

export default function AuthForm({
  isSigningUp,
  handleSubmit,
  toggleIsSigningUp,
}) {
  const [email, setEmail] = useState('')
  const [password, setPassword] = useState('')
  const AuthText = isSigningUp ? 'Sign Up' : 'Log In'

  return (
    <div>
      <h1>{AuthText}</h1>
      <label>
        Email:
        <input value={email} onChange={({ target }) => setEmail(target.value)}/>
      </label>
      <label>
        Password:
        <input type="password" value={password} onChange={({ target }) => setPassword(target.value)} />
      </label>
      <Button handleClick={() => handleSubmit(email, password)} buttonText={AuthText} />
      <div className={styles.linkButton} onClick={toggleIsSigningUp}>
        {isSigningUp ? 'Log In Instead' : 'Sign Up Instead'}
      </div>
    </div>
  )
}
