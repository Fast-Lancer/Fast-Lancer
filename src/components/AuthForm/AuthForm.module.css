.linkButton {
  background: none!important;
  border: none;
  padding-top: 1.2em;
  font-family: 'Work Sans', sans-serif;
  font-weight: 700;
  font-size: 1.2em;
  color: black;
  text-decoration: underline;
  text-underline-offset: .5em;
  cursor: pointer;
}

.authContent {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

}

main {
  border-radius: 20px;
}

.hidden{
  color:white;
}