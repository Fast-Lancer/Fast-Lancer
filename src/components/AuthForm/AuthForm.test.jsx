import { fireEvent, render, screen } from '@testing-library/react'
import { MemoryRouter, Route } from 'react-router-dom'
import AuthForm from './AuthForm.jsx'

it('should match the snapshot', () => {

  const { container } = render(
    <MemoryRouter initialEntries={['/']}>
      <Route path='/'>
        <AuthForm isSigningUp={true} />
      </Route>
    </MemoryRouter>
  )

  expect(container).toMatchSnapshot()
})


it('should call the handleSubimt prop with the entered username and password', async () => {
  let email
  let password
  const mockToggle = jest.fn(() => null)
  const mockSubmit = jest.fn((e, p) => {
    email = e
    password = p
    return null
  })

  const { container } = render(
    <MemoryRouter initialEntries={['/']}>
      <Route path='/'>
        <AuthForm isSigningUp={true} handleSubmit={mockSubmit} />
      </Route>
    </MemoryRouter>

  )

  const emailInput = await screen.findByLabelText('Email:')
  const passwordInput = await screen.findByLabelText('Password:')
  const button = await screen.findByRole('button')

  fireEvent.change(emailInput, { target: { value: '123' } })
  fireEvent.change(passwordInput, { target: { value: 'abc' } })
  fireEvent.click(button)

  expect(mockSubmit).toHaveBeenCalled()
  expect(email).toEqual('123')
  expect(password).toEqual('abc')
})
