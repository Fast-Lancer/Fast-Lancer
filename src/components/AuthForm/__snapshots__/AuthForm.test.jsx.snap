// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div>
  <div
    class="authContent"
  >
    <div
      class="titleBody"
    >
      <section
        class="title"
      >
        Sign Up
      </section>
      <ul
        class="tabnav"
      >
        <li
          class="active"
        >
          <a
            href=" "
          >
            Sign Up
          </a>
        </li>
        <li>
          <a
            href="/"
          >
            ⚛︎
          </a>
        </li>
      </ul>
    </div>
    <main>
      <fieldset>
        <label>
          Email:
          <input
            value=""
          />
        </label>
        <label>
          Password:
          <input
            type="password"
            value=""
          />
        </label>
        <button>
          Sign Up
        </button>
        <div
          class="linkButton"
        >
          Log In Instead
        </div>
      </fieldset>
    </main>
  </div>
</div>
`;
